### PR TITLE
PoC: new viewpager api

### PR DESCRIPTION
**Pull request by MateWW**
# Summary

New API will give you better management of current active page.

- Changed `setPage` and `setPageWithoutAnimation` for `activePage` which accept `number` or `{ page: number, animated: boolean }`
- Removed `initialPage`

### What's required for testing (prerequisites)?
Basic example is updated and using new API

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     |
| Android |    ✅     |

## Checklist

- [x] I have tested this on a device and a simulator
- [ ] I added the documentation in `README.md`
- [x] I updated the typed files (TS and Flow)

### CODE REVIEW

**Comment by oliverdolgener** [comment]
Awesome! I'm looking forward to this! Thank you guys!

File: example/src/App.tsx [hunk 1]
```diff
@@ -83,19 +90,10 @@ export default class ViewPagerExample extends React.Component<{}, State> {
...
   };
-  move = (delta: number) => {
-    const page = this.state.page + delta;
-    this.go(page);
-  };
-
-  go = (page: number) => {
-    if (this.state.animationsAreEnabled) {
-      /* $FlowFixMe we need to update flow to support React.Ref and createRef() */
-      this.viewPager.current?.setPage(page);
```
Comment (enagorny) [inline example/src/App.tsx:94]: this.viewPager ref isn't used anywhere else and could also be deleted

File: src/ViewPager.tsx [hunk 2]
```diff
@@ -57,14 +57,14 @@ export class ViewPager extends React.Component<ViewPagerProps> {
...
 
-  componentDidMount() {
-    // On iOS we do it directly on the native side
-    if (Platform.OS === 'android') {
-      if (this.props.initialPage != null) {
-        this.setPageWithoutAnimation(this.props.initialPage);
-      }
-    }
-  }
+  // componentDidMount() {
```
Comment (enagorny) [inline src/ViewPager.tsx:60]: 🙈

File: src/ViewPager.tsx [hunk 3]
```diff
@@ -139,9 +139,17 @@ export class ViewPager extends React.Component<ViewPagerProps> {
   };
 
   render() {
+    const activePage =
+      typeof this.props.activePage === 'number'
+        ? { page: this.props.activePage, animated: true }
+        : this.props.activePage || { page: 0, animated: true };
+
+    console.log(activePage);
```
Comment (enagorny) [inline src/ViewPager.tsx:147]: 🙈

File: ios/ReactViewPagerManager.m [hunk 4]
```diff
@@ -66,6 +66,18 @@ - (void) changeScrollEnabled
     [self changeScrollEnabled:reactTag enabled:isEnabled];
 }
```
Comment (enagorny) [inline ios/ReactViewPagerManager.m:68]: Should `RCT_EXPORT_METHOD(setPage` and `RCT_EXPORT_METHOD(setPageWithoutAnimation` be deleted?

**Comment by enagorny** [comment]
Looks good 👍

File: ios/ReactViewPagerManager.m (hunk 4 above)
Comment (enagorny) [reply ios/ReactViewPagerManager.m:68]: If we gonna keep it for backward compatibility need to reflect it in Readme

**Pull request closed without merging**